app/actions/GalleryPictureActions.js

```diff
@@ -113,38 +113,43 @@ export function CreateGalleryPicture(galleryPicture: { galleryId: number }) {
     }
   });
 }
-const delay = time => new Promise(res => setTimeout(() => res(), time));
 
-const MAX_UPLOADS = 15;
+const MAX_UPLOADS = 3;
 
-async function uploadGalleryPicturesInTurn(files, galleryId, dispatch) {
-  for (const fileChunk of chunk(files, MAX_UPLOADS)) {
-    await Promise.all(
-      fileChunk.map(file =>
-        dispatch(uploadFile({ file }))
-          .then(action => {
-            if (!action || !action.meta) return;
-            return dispatch(
-              CreateGalleryPicture({
-                galleryId,
-                file: action.meta.fileToken,
-                active: true
-              })
-            );
-          })
-          .catch(error => {
-            dispatch({
-              type: GalleryPicture.UPLOAD.FAILURE,
-              meta: { fileName: file.name }
-            });
-          })
-      )
+function uploadGalleryPicturesInTurn(files, galleryId, dispatch) {
+  const uploadPicture = async file => {
+    const action = await dispatch(uploadFile({ file, timeout: 3 * 60 * 1000 }));
+
+    if (!action || !action.meta) return;
+
+    return dispatch(
+      CreateGalleryPicture({
+        galleryId,
+        file: action.meta.fileToken,
+        active: true
+      })
     );
-    // Delay after each chunk in order to keep CPU and
-    // memory-usage relatively low.
-    // TODO implement a non-pool based approach, with n parallel uploads.
-    await delay(200);
-  }
+  };
+
+  const uploadPictureWithErrorhandler = file =>
+    uploadPicture(file).catch(error => {
+      dispatch({
+        type: GalleryPicture.UPLOAD.FAILURE,
+        error: true,
+        meta: {
+          fileName: file.name,
+          errorMessage: 'Opplasting av bilde feilet.'
+        }
+      });
+    });
+
+  const promiseProducer = function*() {
+    for (const file of files) {
+      yield uploadPictureWithErrorhandler(file);
+    }
+  };
+  const _data = promiseProducer();
```

Comment:
Promise pool should support generator functions, but the reflection done inside the lib doesn't work properly after terser.

Reply:
What happens and why doesn't it work? There is an example in their github using a generator function.
https://github.com/timdp/es6-promise-pool#iterator

Reply:
Yee. It works well without Terser. I think Terser does some shady stuff that changes the type of something. The code crashes (when Terser is enabled) when trying to take `.then` on an Object inside the lib.